spot_driver/test/src/conversions/test_robot_state.cpp

```diff
@@ -0,0 +1,844 @@
+}
+
+TEST(RobotStateConversions, TestGetJointStatesNoJointStates) {
+  // GIVEN a RobotState that does not contain any joint states, but does contains some other kinematic state info
+  ::bosdyn::api::RobotState robot_state;
+  google::protobuf::Timestamp timestamp;
+  timestamp.set_seconds(15);
+  timestamp.set_nanos(0);
+  robot_state.mutable_kinematic_state()->mutable_acquisition_timestamp()->CopyFrom(timestamp);
+
+  // GIVEN some nominal clock skew and prefix
+  google::protobuf::Duration clock_skew;
+  clock_skew.set_seconds(1);
+  constexpr auto prefix = "prefix/";
+
+  // WHEN we create a JointState ROS message from the RobotState
+  const auto out = getJointStates(robot_state, clock_skew, prefix);
+
+  // THEN the message is created and contains a timestamp, but does not contain any joint state data
+  ASSERT_THAT(out.has_value(), IsTrue());
+  EXPECT_THAT(out->header, ClockSkewIsAppliedToHeader(timestamp, clock_skew));
+  EXPECT_THAT(out->name, IsEmpty());
+  EXPECT_THAT(out->position, IsEmpty());
+  EXPECT_THAT(out->velocity, IsEmpty());
+  EXPECT_THAT(out->effort, IsEmpty());
+}
+
+TEST(RobotStateConversions, TestGetJointStatesNoKinematicState) {
+  // GIVEN a RobotState that does not contain any kinematic state data whatsoever
+  ::bosdyn::api::RobotState robot_state;
+
+  // GIVEN some nominal clock skew and prefix
+  google::protobuf::Duration clock_skew;
+  clock_skew.set_seconds(1);
+  constexpr auto prefix = "prefix/";
+
+  // WHEN we attempt to create a JointState ROS message from the RobotState
+  const auto out = getJointStates(robot_state, clock_skew, prefix);
+
+  // THEN no ROS message is output
+  ASSERT_THAT(out.has_value(), IsFalse());
+}
+
+TEST(RobotStateConversions, TestGetTf) {
+  // GIVEN a RobotState containing a valid transform snapshot
+  ::bosdyn::api::RobotState robot_state;
+  // GIVEN a nonzero acquisition timestamp
+  google::protobuf::Timestamp timestamp;
+  timestamp.set_seconds(99);
+  timestamp.set_nanos(0);
+  addAcquisitionTimestamp(robot_state.mutable_kinematic_state(), timestamp);
+  // GIVEN the odom frame is the root of the frame tree
+  addRootFrame(robot_state.mutable_kinematic_state()->mutable_transforms_snapshot(), "odom");
+  // GIVEN the body frame is at a nonzero pose relative to the odom frame
+  addTransform(robot_state.mutable_kinematic_state()->mutable_transforms_snapshot(), "body", "odom", 1.0, 2.0, 3.0, 1.0,
+               0.0, 0.0, 0.0);
+  ASSERT_THAT(bosdyn::api::ValidateFrameTreeSnapshot(robot_state.kinematic_state().transforms_snapshot()),
+              Eq(::bosdyn::api::ValidateFrameTreeSnapshotStatus::VALID));
+
+  // GIVEN some nominal clock skew
+  google::protobuf::Duration clock_skew;
+  clock_skew.set_seconds(1);
+
+  // WHEN we create a TF tree from the RobotState
+  const auto out = getTf(robot_state, clock_skew, "prefix/", "odom");
+  // THEN this succeeds
+  ASSERT_THAT(out.has_value(), IsTrue());
+
+  // THEN the tree contains one frame
```

Comment:
similarly
```suggestion
  // AND THEN the tree contains one frame
```
if you want